### ISSUE TITLE
Add Azure support

### ISSUE THREAD

**Issue by feiskyer** [kubernetes/autoscaler#449]
https://github.com/kubernetes/autoscaler/pull/229 removed Azure support temporarily, we should add Azure support back.

TODOs:

- [x] Bring azure cloud provider back #454 
- [x] Add VMSS support
- [x] Add Availability Set ( standard VM type) support #514
- [x] Multiple node groups #542
- ~~TemplateNodeInfo not supported~~
- [x] Reduce API calls and avoid touching Azure rate limits #552
- [ ] Add AKS support
- [x] Unit tests
- [x] Documentations #641

Depends on: Support VMSS in azure cloud provider https://github.com/kubernetes/kubernetes/issues/43287

**Comment by MaciekPytel** [comment]
This is already discussed at #133 and https://github.com/kubernetes/kubernetes/issues/47511 and some volunteers have posted there. Since none of regular CA maintainers works on Azure, or has any knowledge of it, we fully depend on community for this.

**Comment by feiskyer** [comment]
@MaciekPytel Yep, I'd like to add this back. This will be depending on VMSS feature on Azure.

**Comment by feiskyer** [comment]
Initial Azure support has been added in cluster-autosclaer v1.2.0. Closing this issue.

Will open another issue to track AKS support

**Comment by code4clouds** [comment]
@feiskyer Would post the AKS tracking issue here to follow it up?

**Comment by feiskyer** [comment]
Opened #753 for AKS support
